src/Cake/Arguments/ArgumentParser.cs

```diff
@@ -187,6 +187,11 @@ private bool ParseOption(string name, string value, CakeOptions options)
                 options.Exclusive = ParseBooleanValue(value);
             }
 
+            if (name.Equals("recompile", StringComparison.OrdinalIgnoreCase))
```

Comment:
Suggestion for better name? `invalidate-cache` or similar. Something that refers to cache would be good.

Reply:
Change "recompile" to "regeneratecache"